packages/terra-date-picker/src/DateUtil.js

```diff
@@ -2,34 +2,48 @@
 import moment from 'moment';
 
 class DateUtil {
-  // Converts an ISO 8601 date into a moment object. If the date is invalid and unable to convert, the originally provided date is returned.
+  /**
+   * Creates a moment object using the provided date string.
+   * @param {string|undefined} date - The date to convert. Expect to be in ISO format.
+   * @return {object|undefined} - The moment object. Undefined if unable to convert.
+   */
   static createSafeDate(date) {
-    if (date) {
-      const momentDate = moment(date);
-      return momentDate.isValid() ? momentDate : date;
+    if (!date || (date && date.length === 0)) {
```

Comment:
Empty string is falsey, is there a case where date can be truthy and of length zero?

Reply:
You are correct. The check for length is not needed.
https://github.com/cerner/terra-framework/pull/607/commits/fa9fadf949ff404eb5b8fb2b5a5ca40e414af04b